tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -583,17 +583,181 @@
   </ShaderOp>
 
   <ShaderOp Name="ComputeRawBufferLdStI32" CS="CS">
+    <RootSignature>RootFlags(0), SRV(t0), SRV(t1), UAV(u0), UAV(u1), DescriptorTable(SRV(t2,numDescriptors=2), UAV(u2,numDescriptors=2))</RootSignature>
+    <Resource Name="SRVBuffer0" Dimension="BUFFER" Width="160" InitialResourceState="COPY_DEST" Init="ByName" />
+    <Resource Name="SRVBuffer1" Dimension="BUFFER" Width="160" InitialResourceState="COPY_DEST" Init="ByName" />
+    <Resource Name="SRVBuffer2" Dimension="BUFFER" Width="160" InitialResourceState="COPY_DEST" Init="ByName" />
+    <Resource Name="SRVBuffer3" Dimension="BUFFER" Width="160" InitialResourceState="COPY_DEST" Init="ByName" />
+    <Resource Name="UAVBuffer0" Dimension="BUFFER" Width="480" Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST" TransitionTo="UNORDERED_ACCESS" Init="ByName" Format="R32_SINT" ReadBack="true" />
+    <Resource Name="UAVBuffer1" Dimension="BUFFER" Width="480" Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST" TransitionTo="UNORDERED_ACCESS" Init="ByName" Format="R32_SINT" ReadBack="true" />
+    <Resource Name="UAVBuffer2" Dimension="BUFFER" Width="480" Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST" TransitionTo="UNORDERED_ACCESS" Init="ByName" Format="R32_SINT" ReadBack="true" />
+    <Resource Name="UAVBuffer3" Dimension="BUFFER" Width="480" Flags="ALLOW_UNORDERED_ACCESS" InitialResourceState="COPY_DEST" TransitionTo="UNORDERED_ACCESS" Init="ByName" Format="R32_SINT" ReadBack="true" />
+    <RootValues>
+      <RootValue Index="0" ResName="SRVBuffer0" />
+      <RootValue Index="1" ResName="SRVBuffer1" />
+      <RootValue Index="2" ResName="UAVBuffer0" />
+      <RootValue Index="3" ResName="UAVBuffer1" />
+      <RootValue Index="4" HeapName="ResHeap" />
+    </RootValues>
+    <DescriptorHeap Name="ResHeap" Type="CBV_SRV_UAV">
+      <Descriptor Name='SRVBuffer2' Kind='SRV' ResName='SRVBuffer2' NumElements="10" StructureByteStride="4" />
+      <Descriptor Name='SRVBuffer3' Kind='SRV' ResName='SRVBuffer3' NumElements="10" StructureByteStride="4" />
+      <Descriptor Name='UAVBuffer2' Kind='UAV' ResName='UAVBuffer2' NumElements="30" StructureByteStride="4" />
+      <Descriptor Name='UAVBuffer3' Kind='UAV' ResName='UAVBuffer3' NumElements="30" StructureByteStride="4" />
```

Comment:
Are all of these supposed to be structured buffers, or are some of them supposed to be raw buffers?  If raw, I think you need `Flags="RAW"` and set `NumElements` to the size in DWORD elements, with the `StructureByteStride="0"` or unspecified.  The HLSL accesses even numbered buffers as raw buffers, so I think `"SRVBuffer2"` and `"UAVBuffer2"` need to be changed.  This comment applies to all the cases below.